packages/features/users/components/UserTable/BulkActions/MassAssignAttributes.tsx

```diff
@@ -0,0 +1,279 @@
+                        }
+                        return [...prev, option.id];
+                      });
+                    }
+                  }}>
+                  <span>{option.value}</span>
+                  <div
+                    className={classNames(
+                      "ml-auto flex h-4 w-4 items-center justify-center rounded-sm border"
+                    )}>
+                    {selectedAttributeOption?.includes(option.id) ? (
+                      <Icon name="check" className={classNames("h-4 w-4")} />
+                    ) : null}
+                  </div>
+                </CommandItem>
+              );
+            })}
+          </>
+        ) : (
+          <>
+            <CommandItem>
+              <Input
+                value={selectedAttributeOption[0] || ""}
+                type={foundAttribute.type === "TEXT" ? "text" : "number"}
+                onChange={(e) => {
+                  setSelectedAttributeOption([e.target.value]);
+                }}
+              />
+            </CommandItem>
+          </>
+        )}
+      </CommandGroup>
+    </CommandList>
+  );
+}
+
+export function MassAssignAttributesBulkAction({ table }: Props) {
+  const { selectedAttribute, setSelectedAttribute, foundAttributeInCache } = useSelectedAttributes();
+  const [selectedAttributeOptions, setSelectedAttributeOptions] = useSelectedAttributeOption();
+  const [showMultiSelectWarning, setShowMultiSelectWarning] = useState(false);
+  const { t } = useLocale();
+  const bulkAssignAttributes = trpc.viewer.attributes.bulkAssignAttributes.useMutation({
+    onSuccess: (success) => {
+      showToast(success.message, "success");
+    },
+    onError: (error) => {
+      showToast(`Error assigning attributes: ${error.message}`, "error");
+    },
+  });
+  const { data } = trpc.viewer.attributes.list.useQuery();
+
+  function Content() {
+    if (!selectedAttribute) {
+      return (
+        <>
+          <CommandInput placeholder={t("search")} />
+          <CommandList>
+            <CommandEmpty>No attributes found</CommandEmpty>
+            <CommandGroup>
+              {data &&
+                data.map((option) => {
+                  return (
+                    <CommandItem
+                      key={option.id}
+                      className="hover:cursor-pointer"
+                      onSelect={() => {
+                        setSelectedAttribute(option.id);
+                      }}>
+                      <span>{option.name}</span>
+                      <div
+                        className={classNames("ml-auto flex h-4 w-4 items-center justify-center rounded-sm")}>
+                        <Icon name="chevron-right" className={classNames("h-4 w-4")} />
+                      </div>
+                    </CommandItem>
+                  );
+                })}
+            </CommandGroup>
+          </CommandList>
+        </>
+      );
+    }
+
+    if (showMultiSelectWarning) {
```

Comment:
Warning to show the behaviour of multi select merging with existing options for that attribute.